rust/chains/hyperlane-cosmos/src/providers/grpc.rs

```diff
@@ -268,7 +300,10 @@ impl WasmProvider for WasmGrpcProvider {
     where
         T: Serialize + Send + Sync,
     {
-        self.wasm_query_to(self.contract_address.address(), payload, block_height)
+        let contract_address = self.contract_address.as_ref().ok_or_else(|| {
```

Comment:
IMO since the grpc client insn't really tied to an address, it makes more sense to me to make the address field optional and unwrap it like so. I find it makes for a more intuitive / readable implementation. Could also do this is an separate PR where we apply this change to the other VM providers, lmk

Reply:
I do agree removing `wasm_query` would be a good idea, but there'd be quite a few code shifts so I don't think it's worth it for now